sources_test/path.cpp

```diff
@@ -0,0 +1,163 @@
+#include "utility/file_system/path.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(PathCombine, test_combine_file_3_words)
+{
+
+    std::string path = util::Path::combine<std::string_view>(
+        false, "user", "Desktop", "kus.png");
+
+    EXPECT_STREQ(path.c_str(), "user/Desktop/kus.png");
+}
+TEST(PathCombine, test_combine_folder_3_words)
+{
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    EXPECT_STREQ(path.c_str(), "user/Desktop/kus/");
+}
+
+TEST(PathCombine, test_combine_file_1_word)
+{
+
+    std::string path =
+        util::Path::combine<std::string_view>(false, "us!!dD  Der.png");
+
+    EXPECT_STREQ(path.c_str(), "us!!dD  Der.png");
+}
+TEST(PathCombine, test_combine_folder_1_word)
+{
+    std::string path =
+        util::Path::combine<std::string_view>(true, "useR  WQr/");
+
+    EXPECT_STREQ(path.c_str(), "useR  WQr/");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST(PathGetName, test_get_name_file)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path = util::Path::combine<std::string_view>(
+        false, "user", "Desktop", "kus.png");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus.png");
+
+    std::string name = util::Path::getName(path, context_local);
+
+    EXPECT_STREQ(name.c_str(), "kus.png");
+}
+
+TEST(PathGetName, test_get_name_folder)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus/");
+
+    std::string name = util::Path::getName(path, context_local);
+
+    EXPECT_STREQ(name.c_str(), "kus");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST(PathNormalizeFolderPath, test_normalize_folder_path_with_sep_postfix)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path =
```

Comment:
It is redundant to use `combine` for a `normalizing` test. Either merge tests or remove the 'combine'.